File: src/Tools/Dova.Tools.Java.ClassGenerator/src/main/java/com/github/sejoslaw/dova/MethodDefinitionModel.java
package com.github.sejoslaw.dova;

import java.util.ArrayList;
import java.util.Collection;

public class MethodDefinitionModel {
    public String Modifiers;
    public String ReturnType;
    public String MethodName;
    public Collection<ParameterDefinitionModel> ParameterModels = new ArrayList<>();
    public Collection<TypeParameterModel> TypeParameterModels = new ArrayList<>();
}

File: src/Tools/Dova.Tools.JavaClassStructureGenerator/Models/MethodDefinitionModel.cs
namespace Dova.Tools.JavaClassStructureGenerator.Models;

internal class MethodDefinitionModel
{
    public string Modifiers { get; set; }
    public string ReturnType { get; set; }
    public string MethodName { get; set; }
    public IEnumerable<ParameterDefinitionModel> ParameterModels { get; set; } = new List<ParameterDefinitionModel>();
    public IEnumerable<TypeParameterModel> TypeParameterModels { get; set; } = new List<TypeParameterModel>();
}